sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala

```diff
@@ -199,8 +199,8 @@ class JDBCTableCatalogSuite extends QueryTest with SharedSparkSession {
 
   test("alter table ... rename column") {
     withTable("h2.test.alt_table") {
-      sql("CREATE TABLE h2.test.alt_table (ID INTEGER, C0 INTEGER) USING _")
-      sql("ALTER TABLE h2.test.alt_table RENAME COLUMN ID TO C")
+      sql("CREATE TABLE h2.test.alt_table (id INTEGER, C0 INTEGER) USING _")
+      sql("ALTER TABLE h2.test.alt_table RENAME COLUMN id TO C")
```

Comment:
Without fix (with unquoted column name), this throws the following Exception:
```
Failed table altering: test.alt_table;
org.apache.spark.sql.AnalysisException: Failed table altering: test.alt_table;
	at org.apache.spark.sql.jdbc.JdbcDialect.classifyException(JdbcDialects.scala:268)

Caused by: org.h2.jdbc.JdbcSQLException: Column "ID" not found; SQL statement:
ALTER TABLE "test"."alt_table" RENAME COLUMN id TO "C" [42122-195]
```

Reply:
It should still work if we write `ID`, right? Spark normalizes the columns according to the table schema, so case sensitivity flag can still apply. See `ResolveAlterTableChanges`.

It would be better if we can add tests for case insensitive column resolution.